crates/adapters/dydx/src/websocket/handler.rs

```diff
@@ -0,0 +1,116 @@
+// -------------------------------------------------------------------------------------------------
+//  Copyright (C) 2015-2025 Nautech Systems Pty Ltd. All rights reserved.
+//  https://nautechsystems.io
+//
+//  Licensed under the GNU Lesser General Public License Version 3.0 (the "License");
+//  You may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at https://www.gnu.org/licenses/lgpl-3.0.en.html
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------------------------------
+
+//! Message handler for dYdX WebSocket streams.
+//!
+//! This module processes incoming WebSocket messages and converts them into
+//! Nautilus domain objects.
+
+use ahash::AHashMap;
+use nautilus_model::{identifiers::AccountId, instruments::Instrument};
+use ustr::Ustr;
+
+use super::{
+    DydxWsResult,
+    messages::{DydxWsMessage, NautilusWsMessage},
+};
+
+/// Commands sent to the feed handler.
+#[derive(Debug, Clone)]
+pub enum HandlerCommand {
+    /// Update a single instrument in the cache.
+    UpdateInstrument(Box<nautilus_model::instruments::InstrumentAny>),
+    /// Initialize instruments in bulk.
+    InitializeInstruments(Vec<nautilus_model::instruments::InstrumentAny>),
+}
+
+/// Processes incoming WebSocket messages and converts them to Nautilus domain objects.
+#[derive(Debug)]
+#[allow(dead_code)] // TODO: Remove once implementation is complete
+pub struct FeedHandler {
```

Comment:
Just flagging that a large architectural restructuring is about to land which improves the client <-> handler separation (reduces locking and shared state).